cstore_reader.c

```diff
@@ -200,6 +214,85 @@ CStoreReadFooter(StringInfo tableFooterFilename)
 }
 
 
+/* DeserializeBlockData  deserializes requested data block for all columns. It
+ * uncompresses serialized data if necessary, sets deserialized buffer index at
+ * read state. The function also deallocates previous data buffer used to keep temporary
+ * uncompressed data buffer.
+ */
+static void
+DeserializeBlockData(TupleDesc tupleDescriptor, StripeData *stripeData,
+					 MemoryContext stripeReadContext, uint64 blockIndex,
+					 BlockDataBuffers **blockDataBuffersArray)
+{
+	int columnIndex = 0;
+	MemoryContext oldContext = MemoryContextSwitchTo(stripeReadContext);
+
+	for (columnIndex=0; columnIndex < stripeData->columnCount; columnIndex++)
+	{
+		ColumnData *columnData = stripeData->columnDataArray[columnIndex];
+
+		if (columnData != NULL)
```

Comment:
We usually use ```continue``` for the exceptional cases that can be handled in the beginning of block. I think this is also OK according to Code Complete 2. One advantage of this reducing the indentation levels.

Since the indentation levels are not much in this case, keeping as it is also perfectly fine.